trestle/core/control_io.py

```diff
@@ -807,71 +737,236 @@ def _read_sections(ii: int, lines: List[str], control_id: str,
             control_parts = None
         return ii, control_parts
 
+    @staticmethod
+    def _clean_prose(prose: List[str]) -> List[str]:
+        new_prose = []
+        # remove all known bad lines
+        for line in prose:
+            if not line.startswith(const.SSP_ADD_IMPLEMENTATION_PREFIX) and not line.startswith('___'):
+                new_prose.append(line)
+        # remove empty lines at start and end
+        for _ii, line in enumerate(new_prose):
+            if line.strip():
+                break
+        prose = new_prose[_ii:]
+        for _ii, line in enumerate(reversed(prose)):
+            if line.strip():
+                break
+        clean_prose = prose[:len(prose) - _ii]
+        # if there is no useful prose this will return [''] and allow generation of a statement with empty prose
+        return clean_prose
+
+    @staticmethod
+    def _simplify_name(name: str) -> str:
+        name = name.lower().strip()
+        return re.sub(' +', ' ', name)
+
+    @staticmethod
+    def _comp_name_in_dict(comp_name: str, comp_dict: Dict[str, List[Dict[str, str]]]) -> str:
+        """If the name is already in the dict in a similar form, stick to that form."""
+        simple_name = ControlIOReader._simplify_name(comp_name)
+        for name in comp_dict.keys():
+            if simple_name == ControlIOReader._simplify_name(name):
+                return name
+        return comp_name
+
+    @staticmethod
+    def _add_node_to_dict(
+        comp_name: str, label: str, comp_dict: Dict[str, Dict[str, List[str]]], node: MarkdownNode
+    ) -> None:
+        prose = ControlIOReader._clean_prose(node.content.text)
+        if node.key.startswith('### '):
+            if len(node.key.split()) <= 1:
+                raise TrestleError('Line in control markdown starts with ### but has no component name.')
+            comp_name = node.key.split(' ', 1)[1].strip()
```

Comment:
I think this is something that will be used a lot in the future. Probably worth creating a `MarkdownNode.get_clean_key()` to get key without `###`